File: DatePickerDemo/DatePickerDemo/ViewController.m
//
//  ViewController.m
//  DatePickerDemo
//
//  Created by Guillermo Gruschka on 2/26/16.
//  Copyright © 2016 Guillermo Gruschka. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIDatePicker *datePicker;
@property NSArray *moods;

@end

@implementation ViewController
-(NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView{
    return 1;
}

-(NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component{
    return [[self moods]count];
}

-(NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component{
    return [self moods][row];
}

-(void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component{
    UIColor *moodColor;
    switch (row) {
        case 0:
            moodColor = [UIColor colorWithRed:1.0 green:0.0 blue:0.0 alpha:1.0];
            break;
        case 1:
            moodColor = [UIColor colorWithRed:0.0 green:1.0 blue:0.0 alpha:1.0];
            break;
        case 2:
            moodColor = [UIColor colorWithRed:0.0 green:0.0 blue:1.0 alpha:1.0];
            break;
        default:
            moodColor = [UIColor whiteColor];
            break;
    }
    [[self view] setBackgroundColor:moodColor];
}

- (IBAction)displayDay:(id)sender {
    NSDate *chosenDate = [[self datePicker]date];
    NSDateFormatter *formatter = [[NSDateFormatter alloc]init];
    [formatter setDateFormat:@"EEEE"];
    
    NSString *weekday = [formatter stringFromDate:chosenDate];
    NSString *msg = [[NSString alloc] initWithFormat:@"That's a %@", weekday];
    
    UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"What day is that?" message:msg delegate:nil cancelButtonTitle:@"Okay." otherButtonTitles:nil, nil];
    
    [alert show];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setMoods:@[@"Happy",@"Sad",@"Angry",@"Love"]];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: TicTacToe/TicTacToe/ViewController.swift
//
//  ViewController.swift
//  TicTacToe
//
//  Created by Guillermo Gruschka on 5/6/16.
//  Copyright © 2016 Guillermo Gruschka. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var btn1: UIButton!
    @IBOutlet weak var btn2: UIButton!
    @IBOutlet weak var btn3: UIButton!
    @IBOutlet weak var btn4: UIButton!
    @IBOutlet weak var btn5: UIButton!
    @IBOutlet weak var btn6: UIButton!
    @IBOutlet weak var btn7: UIButton!
    @IBOutlet weak var btn8: UIButton!
    @IBOutlet weak var btn9: UIButton!
    @IBOutlet weak var turnLabel: UILabel!
    @IBOutlet weak var finnMovesLabel: UILabel!
    @IBOutlet weak var jakeMovesLabel: UILabel!
    
    var turn = 0
    let winnerPlays = [[1,2,3],[4,5,6],[7,8,9],[1,5,9],[3,5,7]]
    var firstPlayerPlays: [Int] = []
    var secondPlayerPlays: [Int] = []
    var plays: [Int] = []

    @IBAction func btnClick(sender: AnyObject) {
        if (!plays.contains(sender.tag)){
            switch turn {
            case 0:
                sender.setImage(UIImage(named: "finn.png"), forState: .Normal)
                turn += 1
                turnLabel.text = "Jake's turn"
                plays.append(sender.tag!)
                firstPlayerPlays.append(sender.tag!)
            case 1:
                sender.setImage(UIImage(named: "jake.png"), forState: .Normal)
                turn = 0
                turnLabel.text = "Finn's turn"
                plays.append(sender.tag!)
                secondPlayerPlays.append(sender.tag!)
            default:
                break
            }
        }
        
        finnMovesLabel.text = "Finn's moves: " + firstPlayerPlays.description
        jakeMovesLabel.text = "Jake's moves: " + secondPlayerPlays.description
        

    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        if turn == 0{
            turnLabel.text = "Finn's turn"
        } else{
            turnLabel.text = "Jake's turn"
        }
        
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

